### ISSUE TITLE
[bert/pytorch] the keys of pretrained weights can not match the keys of cased base model

### ISSUE THREAD

**Issue by shunshen93** [NVIDIA/DeepLearningExamples#672]
I have loaded the checkpoint for bert-base-cased, then finetine on SQuAD v1.1, but can not reproduce the Accuracy - FP32 F1.

It is the keys of pretrained weights can not match the keys of cased base model

the model keys is as follow:

'bert.embeddings.word_embeddings.weight', 'bert.embeddings.position_embeddings.weight', 'bert.embeddings.token_type_embeddings.weight', 'bert.embeddings.LayerNorm.weight', 'bert.embeddings.LayerNorm.bias', 'bert.encoder.layer.0.attention.self.query.weight', 'bert.encoder.layer.0.attention.self.query.bias', 'bert.encoder.layer.0.attention.self.key.weight', 'bert.encoder.layer.0.attention.self.key.bias', 'bert.encoder.layer.0.attention.self.value.weight', 'bert.encoder.layer.0.attention.self.value.bias', 'bert.encoder.layer.0.attention.output.dense.weight', 'bert.encoder.layer.0.attention.output.dense.bias', 'bert.encoder.layer.0.attention.output.LayerNorm.weight', 'bert.encoder.layer.0.attention.output.LayerNorm.bias', 'bert.encoder.layer.0.intermediate.dense_act.weight', 'bert.encoder.layer.0.intermediate.dense_act.bias'
...

the checkpoint keys is as follow:

'bert.embeddings.word_embeddings.weight', 'bert.embeddings.position_embeddings.weight', 'bert.embeddings.token_type_embeddings.weight', 'bert.embeddings.LayerNorm.weight', 'bert.embeddings.LayerNorm.bias', 'bert.encoder.layer.0.attention.self.query.weight', 'bert.encoder.layer.0.attention.self.query.bias', 'bert.encoder.layer.0.attention.self.key.weight', 'bert.encoder.layer.0.attention.self.key.bias', 'bert.encoder.layer.0.attention.self.value.weight', 'bert.encoder.layer.0.attention.self.value.bias', 'bert.encoder.layer.0.attention.output.dense.weight', 'bert.encoder.layer.0.attention.output.dense.bias', 'bert.encoder.layer.0.attention.output.LayerNorm.weight', 'bert.encoder.layer.0.attention.output.LayerNorm.bias', 'bert.encoder.layer.0.intermediate.dense.weight', 'bert.encoder.layer.0.intermediate.dense.bias'
...

it can not match in the intermediate layer, and it cause that the f1 score is lower than 80.
when I changed the intermediate layer of model, then the F1 score is right

**Comment by sharathts** [comment]
Our checkpoints and code [fuse linear transform with activations](https://github.com/NVIDIA/DeepLearningExamples/blob/master/PyTorch/LanguageModeling/BERT/modeling.py#L142). Hence the reason for the mismatch. 
To avoid this, model has to be an instance of BertModel as in the link above, or you can write a script that renames `dense_act` to `dense` - based on your requirements,.

**Comment by sharathts** [comment]
Closing the bug. Feel free to reopen if you have more questions.

**Comment by sharathts** [comment]
Simpler than writing a script, you can also add weight mapping here: https://github.com/NVIDIA/DeepLearningExamples/blob/master/PyTorch/LanguageModeling/BERT/modeling.py#L725

            if 'intermediate.dense.' in key:
                new_key = key.replace('intermediate.dense.', 'intermediate.dense_act.')
We also provide a bert-base-uncased checkpoint on NGC
Closing the bug. Feel free to reopen if you have more questions.
